### PR TITLE
Implement the new HtmlFomatter to write the output to html file

### PR DESCRIPTION
**Pull request by gouniLee**
Implemnt the HtmoFormatter to write the output into a html file.
related change: 
* ilib-lint: https://github.com/iLib-js/ilib-lint/pull/93
* ilib-lint-common: https://github.com/iLib-js/ilib-lint-common/pull/16
* ilib-samples: https://github.com/iLib-js/ilib-samples/pull/24

### CODE REVIEW

File: src/HtmlFormatter.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        let end = '</body>\n' +
+                  '</html>\n';
+        return header + resultList + end;
+    }
+    writeSummaryToFile(prjName, totaltime, fileStats, resultStats, score) {
+        const fmt = new Intl.NumberFormat("en-US", {
+            maxFractionDigits: 2
+        });
+
+        let summaryTable = '<p style="color:#714AFF;text-align:left;font-size:30px;font-weight:bold" width=300px;> [' + prjName + '] Sumarry</p>\n'+
```
Comment (ehoogerbeets) [inline src/HtmlFormatter.js]: sp. Summarry -> Summary

File: src/HtmlFormatter.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                           '<table>\n' +
+                           '<thead>\n' +
+                           '<tr><td style="font-size:20px">Total Elapse Time</td>\n' +
```
Comment (ehoogerbeets) [inline src/HtmlFormatter.js]: Elapse -> Elapsed

File: src/HtmlFormatter.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                           '</table>\n' +
+                           '<br/>\n' +
+                           '<hr align="left" width="1100px"/>\n' +
+                           '<p style="color:#714AFF;text-align:left;font-size:30px;font-weight:bold" width=320px; >Detailed Information</p>\n';
+        return summaryTable;
+    }
+
+    format(result) {
+        let levelTextColor = (result.severity === "error") ? "color:red;" : "color:orange;";
+        let autofix = (result.fix === undefined) ? "unavilable" : result.fix.applied;
```
Comment (ehoogerbeets) [inline src/HtmlFormatter.js]: unavilable -> unavailable

File: src/HtmlFormatter.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+        let htmlText = '<table>\n' +
+                       '<thead>\n' +
+                       '<tr><th style=' + levelTextColor + "text-align:left;font-size:22px; width=280px;>" + "["+result.severity + "]"+ "</th><th style='text-align:left'></th></tr>\n" +
+                       '<tr><td style="text-align:left;color:Cadetblue;font-weight:bold">filepath</td><td style="color:Cadetblue;font-weight:bold">' + result.pathName + "</td></tr>\n" +
```
Comment (ehoogerbeets) [inline src/HtmlFormatter.js]: Should put the location information in here as well to make it easier to find the problems in the original file.

File: src/HtmlFormatter.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                       '<tr><td style="color:red; text-align:left">Descriptions</td><td style="color:red;" >' + result.description + "</td></tr>\n" +
+                       '<tr><td>key' + "</td><td>" + result.source +  "</td></tr>\n" +
+                       '<tr><td>source' + "</td><td>" + result.id +  "</td></tr>\n" +
+                       '<tr><td>target' + "</td><td>" + result.highlight + "</td></tr>\n" +
```
Comment (ehoogerbeets) [inline src/HtmlFormatter.js]: Should make the highlights visible:

```javascript
... + result.highlight.replaceAll(/<e\d>/g, '<span style="color:red">').replaceAll(/<\/e\d>/g, '</span>') + ...
```

File: src/HtmlFormatter.js (hunk 1 above)
Comment (gouniLee) [reply src/HtmlFormatter.js]: fixed

File: src/HtmlFormatter.js (hunk 2 above)
Comment (gouniLee) [reply src/HtmlFormatter.js]: fixed

File: src/HtmlFormatter.js (hunk 3 above)
Comment (gouniLee) [reply src/HtmlFormatter.js]: fixed

File: src/HtmlFormatter.js (hunk 5 above)
Comment (gouniLee) [reply src/HtmlFormatter.js]: fixed

File: src/HtmlFormatter.js (hunk 4 above)
Comment (gouniLee) [reply src/HtmlFormatter.js]: Regarding the linking of the real file path, I'll work on it different PR

**Comment by gouniLee** [comment]
@ehoogerbeets  I changed the status from the draft. Please review them again.

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            npm run test
+
+workflows:
+  version: 2
+  test-all-node-versions:
+    jobs:
+      - test:
+          docker_image: circleci/node:14-browsers
+      - test:
+          docker_image: circleci/node:16-browsers
```
Comment (ehoogerbeets) [inline .circleci/config.yml]: Should test on later versions of node as well. Also, need to use the "modern" circleci images, as eventually, these old ones are going away. See https://github.com/iLib-js/ilib-lint-common/blob/main/.circleci/config.yml as an example

File: README.md [hunk 7]
```diff
@@ -1,2 +1,30 @@
...
+## License
+
+Copyright (c) 2024, JEDLSoft
+
+This plugin is license under Apache2. See the [LICENSE](./LICENSE)
+file for more details.
+
+## Release Notes
+### v1.0.0
+- Implemnt the HtmoFormatter to write the output into a html file.
```
Comment (ehoogerbeets) [inline README.md]: Implement
HtmlFormatter

File: src/HtmlFormatter.js [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+ */
+class HtmlFormatter extends Formatter {
+    /**
+     * Construct an formatter instance.
+     * @constructor
+     */
+    constructor(options) {
+        super(options);
+        this.name = "html-formatter";
+        this.description = "Formats results for an html file with colors.";
```
Comment (ehoogerbeets) [inline src/HtmlFormatter.js]: "Formats results for an html file with colors." -> "Formats results as an html file with colors."

File: src/HtmlFormatter.js [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+                     '</head>\n' +
+                     '<body>\n';
+        return header;
+    }
+    _formatFooter(){
+        let end = '</body>\n' +
+                  '</html>\n';
+        return end;
+    }
+    _formatResult(results, errorsOnly) {
```
Comment (ehoogerbeets) [inline src/HtmlFormatter.js:115]: Put a comment with @private for these ones so they don't get documented

**Review by ehoogerbeets — APPROVED** [review]

**Pull request merged**